docs/conf.py

```diff
@@ -146,3 +148,102 @@
+            # because every function needs a docstring
+            if len(lines) == 0:
+                # raise Exception('No docstring provided for function %s' % name)
+                return
+
+            # all docstrings need a description, if we're getting into the args list immediately
+            # that is a bad, bad thing
+            if len(lines) > 0 and lines[0][0] == ':':
+                raise Exception('No description before args list given for %s' % name)
+
+            # the first value of lines should always contain the start of the description
+            # if there is an extra space in front, that violates how a Google doctring should look
+            if lines[0][0].isspace():
+                raise Exception('Your description for %s should not have any preceding whitespace' % name)
+
+            # handle the Args section, all args should have an associated :param and :type in the lines list
+            param_args = [re.match(r':param (\S*):', line).group(1) for line in lines if re.match(r':param (\S*):', line)]
+            type_args = [re.match(r':type (\S*):', line).group(1) for line in lines if re.match(r':type (\S*):', line)]
+
+            # usually this happens when the person writing the docs does not know how to tab properly
+            # and ReadTheDocs gets screwed over when processing so reads something
+            # in an argument description as an actual argument
+            if sorted(param_args) != sorted(type_args):
+                raise Exception('Parameter list: %s and type list: %s do not match in %s, a formatting error in your Args section likely caused this' % (','.join(param_args), ','.join(type_args), name))
+
+            # usually if the user forgets to specify an argument, occassionally a formatting issue as well (ex. with Returns)
+            if len(param_args) < len(argnames):
+                missing_args = list(set(argnames).difference(set(param_args)))
+                raise Exception('Missing description for parameters %s in %s, check your Args list and your docstring formatting' % (','.join(missing_args), name))
+
+            # usually if the user adds an extra argument, occassionally a formatting issue as well
+            if len(param_args) > len(argnames):
+                extra_args = list(set(param_args).difference(set(argnames)))
+                raise Exception('Extraneous arguments specified: %s in %s, check your Args list and your docstring formatting' % (','.join(extra_args), name))
+
+            # added just in case the checks above somehow missed something
+            # this might be the only one we really need but I think the others
+            # beforehand are a bit more descriptive
+            if sorted(param_args) != sorted(argnames):
+                raise Exception('Parameter list: %s does not match arglist: %s in %s, check your docstring formatting' % (','.join(param_args), ','.join(argnames), name))
```

Comment:
I think we just need the last check; the two above are encompassed by this one and I don't think being explicit about the type of args mismatch is needed if we're printing out each list anyways.